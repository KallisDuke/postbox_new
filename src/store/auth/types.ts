import {Email, User, Credentials} from '../../types';
export const GET_EMAILS_SUCCESS = 'GET_EMAILS_SUCCESS'
export const START_LOGIN = 'START_LOGIN'
export const LOGIN_SUCCESS = 'START_LOGIN_SUCCESS';
export const LOGIN_ERROR= 'START_LOGIN_ERROR';
export const START_SIGNOUT = 'START_SIGNOUT';
export const SIGNOUT_SUCCESS = 'SIGNOUT_SUCCESS';

interface signInAction {
	type: typeof START_LOGIN,
	credentials : Credentials   
}

interface signInSuccess {
	type: typeof LOGIN_SUCCESS,
	currentUser : User   
}

interface signOutAction {
	type : typeof START_SIGNOUT,
	currentUser : User
}

interface signOutSuccess {
	type : typeof SIGNOUT_SUCCESS,
}

export type ActionTypes = signInAction | signInSuccess | signOutAction | signOutSuccess;
