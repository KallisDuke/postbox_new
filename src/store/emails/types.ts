import {Email, EmailCreation} from '../../types';
export const GET_EMAILS_SUCCESS = 'GET_EMAILS_SUCCESS'
export const GET_EMAILS_START = 'GET_EMAILS_START'
export const SEND_EMAIL_START = 'SEND_EMAIL_START';
export const SEND_EMAIL_SUCCESS = 'SEND_EMAIL_SUCCESS';
export const RESET_OPERATIONS = 'RESET_OPERATIONS';
export const DELETE_EMAIL = 'DELETE_EMAIL';
export const DELETE_EMAIL_SUCCESS = 'DELETE_EMAIL_SUCCESS';
export const SET_EMAIL_READ = 'SET_EMAIL_READ';

interface getEmailsAction {
		type: typeof GET_EMAILS_START,
		email: string,
		emailAction?:string    
}

interface getEmailsSuccessAction {
	type: typeof GET_EMAILS_SUCCESS,
	emails: Email[]    
}

interface sendEmailAction {
	type : typeof SEND_EMAIL_START,
	payload : EmailCreation 
}

interface sendEmailSuccess {
	type : typeof SEND_EMAIL_SUCCESS,
	message : string 
}

interface resetOperationAction {
	type : typeof RESET_OPERATIONS
}

interface deleteEmail {
	type : typeof DELETE_EMAIL,
	emailUuids : string[]
}

interface deleteEmailSuccess {
	type : typeof DELETE_EMAIL_SUCCESS,
	emails: Email[]    
}

interface setEmailRead {
	type : typeof SET_EMAIL_READ,
	emailUuid : string
}

export type ActionTypes = getEmailsAction | getEmailsSuccessAction | sendEmailAction | sendEmailSuccess | resetOperationAction | deleteEmail | deleteEmailSuccess | setEmailRead;
