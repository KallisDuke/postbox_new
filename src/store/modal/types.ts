export const CLOSE_MODAL = 'CLOSE_MODAL';
export const OPEN_MODAL = 'OPEN_MODAL';

export interface ModalProps {
    type : string,
    header: string,
    className:string,
    open : boolean
}

interface openModalAction {
    type: typeof OPEN_MODAL,
    data : ModalProps,
}

interface closeModalAction {
    type: typeof CLOSE_MODAL,
    // data : ModalProps,
}



export type ActionTypes = openModalAction | closeModalAction;
