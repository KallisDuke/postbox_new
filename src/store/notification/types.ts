export const RESET_OPERATIONS = 'RESET_OPERATIONS';
export const SEND_NOTIFICATION = 'SEND_NOTIFICATION';
export const SEND_ERROR_NOTIFICATION = 'SEND_ERROR_NOTIFICATION';
interface notifyAction {
    type: typeof SEND_NOTIFICATION,
    message : string,
}

interface notifyErrorAction {
    type: typeof SEND_ERROR_NOTIFICATION,
    message : string,
}

interface resetAction {
	type : typeof RESET_OPERATIONS
}

export type ActionTypes = resetAction | notifyAction | notifyErrorAction;
