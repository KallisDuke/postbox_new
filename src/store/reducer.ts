import { combineReducers } from 'redux';
import {emailReducer} from './emails/reducer';
import {modalReducer} from './modal/reducer';
import {notificationReducer} from './notification/reducer';
import {authReducer} from './auth/reducer';

const rootReducer = combineReducers({
    email: emailReducer,
    modal : modalReducer,
    notification : notificationReducer,
    auth: authReducer
    // connection : connectionReducer
})

export default rootReducer;
