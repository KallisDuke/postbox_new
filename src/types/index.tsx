export interface EmailCreation {
	to: string[],
	cc: string[],
	body: string,
	subject: string,
	sender:string
}
export interface Email extends EmailCreation {
	read: boolean,
	created_at: string,
}

export interface User {
	id: number,
	name: string,
	email: string,
	password : string
}

export interface Credentials {
	email: string,
	password: string
}

export interface formattedEmails {
	sender_email : string,
	sender_name : string,
	body: string,
	subject: string,
	read: boolean,
	created_at: string,
	emailUuid : string
}

export interface ParamTypes {
	emailAction: string
	emailUuid?:string
}
