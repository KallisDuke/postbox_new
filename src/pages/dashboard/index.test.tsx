import React from 'react';
import { render } from '@testing-library/react';
import Dashboard from './index';
import {dashboardProps} from '../../types';
import {configureStore} from '../../store';
import { Provider } from "react-redux";
test('renders dashboard', () => {
  const store = configureStore();
  
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    }))
  });

  const { getByText } = render(
    <Provider store={store}>
      <div className="App">
        <Dashboard />
      </div>
    </Provider>
  );
  // const linkElement = getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
