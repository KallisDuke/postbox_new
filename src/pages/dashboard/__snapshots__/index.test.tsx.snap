// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should match its own snapshot 1`] = `null`;
